wrapanapi/systems/ec2.py

```diff
@@ -36,14 +31,17 @@ def _regions(regionmodule, regionname):
 
 class _TagMixin(object):
     def set_tag(self, key, value):
-        self.system.api.create_tags([self.uuid], tags={key: value})
+        self.system.ec2_connection.create_tags(Resources=[self.uuid], Tags={key: value})
 
     def get_tag_value(self, key):
         self.refresh()
-        return self.raw.tags.get(key)
+        for tag in self.raw.tags:
+            if tag.get('Key') == key:
+                return tag.get('Value')
```

Comment:
Hit an exception here while trying to access template names:

```
In [18]: [t.name for t in e.mgmt.list_templates()]
returning cached mgmt class for 'ec2west'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-18-9d477b10f96c> in <module>()
----> 1 [t.name for t in e.mgmt.list_templates()]

/home/mshriver/repos/wrapanapi/wrapanapi/systems/ec2.py in name(self)
    301     @property
    302     def name(self):
--> 303         tag_value = self.get_tag_value('Name')
    304         return self.raw.name or tag_value if tag_value else self.raw.id
    305 

/home/mshriver/repos/wrapanapi/wrapanapi/systems/ec2.py in get_tag_value(self, key)
     36     def get_tag_value(self, key):
     37         self.refresh()
---> 38         for tag in self.raw.tags:
     39             if tag.get('Key') == key:
     40                 return tag.get('Value')

TypeError: 'NoneType' object is not iterable
```

Reply:
Fixed.